src/crypto/box_/curve25519xsalsa20poly1305.rs

```diff
@@ -117,6 +147,33 @@ pub fn open(c: &[u8],
     }
 }
 
+/// `open_detached()` verifies and decrypts a ciphertext `c` using the receiver's secret key `sk`,
+/// the senders public key `pk`, and a nonce `n`. `c` is decrypted in place, so if this function is
+/// successful it will contain the plaintext. If the ciphertext fails verification,
+/// `open_detached()` returns `Err(())`, and the ciphertext is not modified.
+pub fn open_detached(c: &mut [u8],
+                     mac: &Tag,
+                     &Nonce(ref n): &Nonce,
+                     &PublicKey(ref pk): &PublicKey,
+                     &SecretKey(ref sk): &SecretKey) -> Result<(), ()> {
```

Comment:
The change proposed above would also improve the ergonomics of the return value here. So that the return value would be `Result<&[u8], ()>`. (by the way I really need to get to adding real error types for sodiumoxide).